### ISSUE TITLE
Access to phone contacts to pick stored address as location for new calendar entry

### ISSUE THREAD

**Issue by hhartzer** [FossifyOrg/Calendar#69]
Migrated from: https://github.com/SimpleMobileTools/Simple-Calendar/issues/1929

"Hello,
This is a feature request to add the functionality to pick an address stored under a contact in the phone book to use it as the location for a new calendar entry.
This is a standard feature in nearly all calendar apps. I am missing it in simple calendar, unfortunately.

I have several recurring calendar entries for which I have also a phone contact including the address. To be able to pick it right away from the contact details would be highly convenient."

**Comment by Aga-C** [comment]
Please adjust your feature request report to the [issue template](https://raw.githubusercontent.com/FossifyOrg/Voice-Recorder/master/.github/ISSUE_TEMPLATE/feature_request.md).

**Comment by froedel** [comment]
> Please adjust your feature request report to the [issue template](https://raw.githubusercontent.com/FossifyOrg/Voice-Recorder/master/.github/ISSUE_TEMPLATE/feature_request.md).

The link is dead. Please check.

**Comment by Ionshard** [comment]
I am also here from the SMT issue. @froedel Here is the issue template filled out to the best of my knowledge, feel free to replace the original issue with the following and change anything you think should be updated.

---

```
### Checklist

    * [x]  I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Calendar/issues) or [closed](https://github.com/FossifyOrg/Calendar/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
    * [x]  I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Calendar/discussions) or [closed](https://github.com/FossifyOrg/Calendar/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
    * [x]  I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
    * [x]  **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
    * [x]  This issue contains only one feature request.
    * [x]  I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Calendar/blob/master/CONTRIBUTING.md).
    * [x]  I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

This is a feature request to add the functionality to pick an address stored under a contact in the phone book to use it as the location for a new calendar entry.

This is a standard feature in nearly all calendar apps. I am missing it in simple calendar, unfortunately.

I have several recurring calendar entries for which I have also a phone contact including the address. To be able to pick it right away from the contact details would be highly convenient.

### Why do you want this feature?

To make it easier to enter location information into a calendar event.

### Additional information

Since the Calendar application already requests contact permission when adding attendees this information should hopefully already be available in app. Furthermore fetching the location data from the contacts would be a good way to make it easier to add location information to a calendar event without needing to access the internet to do a location lookup.
```

**Comment by Ionshard** [comment]
For my own representation from the original thread:

> I came here to request the same feature that @froedel suggested and I am not sure how @dumblob is suggesting would work for this so I just want to clarify.
> 
> For example I just booked a checkup with my doctor. I have my clinic stored in my contacts and I have their address stored with the contact. When I go to add an event to my calendar via Google Calendar when I enter the `Add location` field it takes me to a search bar where I can search "clinic" and it will find the entry from my contacts and show it as a valid address that I can select. When I select that contact, the address stored in the contact is then used as the location for the event.
> 
> I would love to see this with Simple Calendar. When creating a new event I can search my contacts for an address to use as the location of the event rather than needing to enter the address manually.
> 
> This is important to me because I also have Waze reading from my calendar to suggest times that I leave in order to arrive on time for my events, so being able to effectively add the location to my events is very important so I am not late due to things like rush hour traffic!
> 
> Thank you for such a great app!

**Comment by Aga-C** [comment]
@Ionshard Please create properly a new issue for that, and then I'll close this one, since original author didn't reply.

**Comment by Ionshard** [comment]
@Aga-C ... will do ... but I am sad about losing out on issue #69

**Comment by hhartzer** [comment]
Thank you @Ionshard for filling out the template! I updated this issue with it. Sorry for the wait on this.

**Comment by Aga-C** [comment]
@hhartzer Thanks for clarifying the issue. Now there is no need to create another one.

**Comment by Ionshard** [comment]
I was just collecting reference images for the new issue, glad we didn't have to fragment the issue. I will finish the images and post them here.

**Comment by Ionshard** [comment]
Here is how this works using Google Calendar. This is provided as an example only, the feature request does not need to be implemented in this exact manner. It is just showing how the desired feature is accomplished in Google Calendar.

1. When you go to create a new calendar entry there is a section for Location:

![Google Calendar - Create Event](https://github.com/FossifyOrg/Calendar/assets/86419/212d1aa9-5239-4352-ab95-a567443f209e)

2. If you select the location it takes you to a search form:
![Google Calendar - Location Search](https://github.com/FossifyOrg/Calendar/assets/86419/b245ca1d-9d41-4675-b7a5-ca28f528ad82)

3. If I enter in the partial name of a contact (in this case `VCA Alta`) then it finds entries in my phone's contact list for matching entries with the address stored for that contact. In this case it finds the `VCA Alta Vista Animal Hospital` contact. (Ignore the portion scratched out in red, Google also searches the internet for the term to find nearby places. That is **not** part of this request. Only the local information from the contacts is requested)
![Google Calendar - Searched Locations](https://github.com/FossifyOrg/Calendar/assets/86419/224c38bb-d009-47d3-ae37-4325dd1d2102)

4. When the contact is selected, the address for that contact is entered as the location in the calendar entry:
![Google Calendar - Address Selected](https://github.com/FossifyOrg/Calendar/assets/86419/93443157-4938-4249-bcd0-9c57d1e90bad)

Something simillar in the Fossify Calendar would be great, again because I already gave Fossify Calendar app my contact permission for attendees and this wouldn't require internet access at all which was a core tenant of SMT's Calendar.

**Comment by froedel** [comment]
Thank you @lonshard for documenting it.

**Comment by dumblob** [comment]
Thank you @Ionshard for the details! Indeed, my suggestion to duplicate an existing calendar entry would not cut it here. Your proposal on the other hand seems to cover what is being asked for by @hhartzer.
